test/Microsoft.AspNetCore.Mvc.FunctionalTests/BasicTests.cs

```diff
@@ -244,25 +248,62 @@ public BasicTests(MvcTestFixture<BasicWebSite.Startup> fixture)
         }
 
         [Fact]
-        public async Task JsonHelperWithSettings_RendersJson()
+        public async Task JsonHelperWithSettings_RendersPascalCaseJson()
         {
             // Arrange
+            var settings = JsonSerializerSettingsProvider.CreateSerializerSettings();
+            settings.ContractResolver = new DefaultContractResolver
+            {
+                NamingStrategy = new DefaultNamingStrategy(),
+            };
             var json = JsonConvert.SerializeObject(
                 new BasicWebSite.Models.Person()
                 {
                     Id = 9000,
                     Name = "John <b>Smith</b>"
                 },
-                new JsonSerializerSettings { ContractResolver = new CamelCasePropertyNamesContractResolver() });
+                settings);
+            var expectedBody = string.Format(
+                @"<script type=""text/javascript"">
+    var json = {0};
+</script>",
+                json);
+
+            // Act
+            var response = await Client.GetAsync("Home/JsonHelperWithSettingsInView?snakeCase=false");
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            Assert.Equal("text/html", response.Content.Headers.ContentType.MediaType);
+
+            var actualBody = await response.Content.ReadAsStringAsync();
+            Assert.Equal(expectedBody, actualBody, ignoreLineEndingDifferences: true);
+        }
 
+        [Fact]
+        public async Task JsonHelperWithSettings_RendersSnakeCaseJson()
+        {
+            // Arrange
+            var settings = JsonSerializerSettingsProvider.CreateSerializerSettings();
+            settings.ContractResolver = new DefaultContractResolver
+            {
+                NamingStrategy = new SnakeCaseNamingStrategy(),
+            };
+            var json = JsonConvert.SerializeObject(
+                new BasicWebSite.Models.Person()
+                {
+                    Id = 9000,
```

Comment:
It would be better if the properties have a more complex name that demonstrate the different encodings, like PersonName or FirstName or something like that, so that different naming strategies are clear.